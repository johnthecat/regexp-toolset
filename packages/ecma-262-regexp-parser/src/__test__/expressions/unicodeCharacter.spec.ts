import { expect, describe, it } from 'vitest';
import { parseRegexp } from '../../index.js';

describe('Octal char', () => {
  it('should correctly unicode value', () => {
    expect(parseRegexp(/\u0061/)).toMatchSnapshot();
  });

  it('should fallback to char sequence, if octal value is out of range', () => {
    // Largest unicode value - ffff₁₆ (65535₁₀)
    expect(parseRegexp(/\ufffg/)).toMatchSnapshot();
  });

  it('should fallback to char sequence, if there is not enough chars', () => {
    expect(parseRegexp(/\u006\a/)).toMatchSnapshot('with char at the end');
    expect(parseRegexp(/\u006/)).toMatchSnapshot('with regexp end at the end');
  });
});
