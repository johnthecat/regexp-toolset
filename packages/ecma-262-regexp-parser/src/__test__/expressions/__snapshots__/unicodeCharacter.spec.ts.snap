// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Octal char > should correctly unicode value 1`] = `
{
  "body": {
    "charCode": 97,
    "end": 6,
    "kind": 6,
    "start": 1,
    "type": "unicode",
    "value": "a",
  },
  "end": 7,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;

exports[`Octal char > should fallback to char sequence, if octal value is out of range 1`] = `
{
  "body": {
    "end": 5,
    "expressions": [
      {
        "end": 2,
        "kind": 8,
        "start": 1,
        "type": 4,
      },
      {
        "charCode": 102,
        "end": 3,
        "kind": 6,
        "start": 3,
        "type": "simple",
        "value": "f",
      },
      {
        "charCode": 102,
        "end": 4,
        "kind": 6,
        "start": 4,
        "type": "simple",
        "value": "f",
      },
      {
        "charCode": 100,
        "end": 5,
        "kind": 6,
        "start": 5,
        "type": "simple",
        "value": "d",
      },
    ],
    "kind": 5,
    "start": 1,
  },
  "end": 6,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;

exports[`Octal char > should fallback to char sequence, if there is not enough chars > with char at the end 1`] = `
{
  "body": {
    "end": 7,
    "expressions": [
      {
        "charCode": 117,
        "end": 2,
        "kind": 6,
        "start": 1,
        "type": "escaped",
        "value": "u",
      },
      {
        "charCode": 48,
        "end": 3,
        "kind": 6,
        "start": 3,
        "type": "simple",
        "value": "0",
      },
      {
        "charCode": 48,
        "end": 4,
        "kind": 6,
        "start": 4,
        "type": "simple",
        "value": "0",
      },
      {
        "charCode": 54,
        "end": 5,
        "kind": 6,
        "start": 5,
        "type": "simple",
        "value": "6",
      },
      {
        "charCode": 97,
        "end": 7,
        "kind": 6,
        "start": 6,
        "type": "escaped",
        "value": "a",
      },
    ],
    "kind": 5,
    "start": 1,
  },
  "end": 8,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;

exports[`Octal char > should fallback to char sequence, if there is not enough chars > with regexp end at the end 1`] = `
{
  "body": {
    "end": 5,
    "expressions": [
      {
        "charCode": 117,
        "end": 2,
        "kind": 6,
        "start": 1,
        "type": "escaped",
        "value": "u",
      },
      {
        "charCode": 48,
        "end": 3,
        "kind": 6,
        "start": 3,
        "type": "simple",
        "value": "0",
      },
      {
        "charCode": 48,
        "end": 4,
        "kind": 6,
        "start": 4,
        "type": "simple",
        "value": "0",
      },
      {
        "charCode": 54,
        "end": 5,
        "kind": 6,
        "start": 5,
        "type": "simple",
        "value": "6",
      },
    ],
    "kind": 5,
    "start": 1,
  },
  "end": 6,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;
